server/api.js

```diff
@@ -77,10 +77,53 @@ async function getSensor(req, res) {
   }
 }
 
+const validateMessageClients = Validator.body(['braveKey', 'message']).exists()
+
+async function messageClients(req, res) {
+  const validationErrors = Validator.validationResult(req).formatWith(helpers.formatExpressValidationErrors)
+
+  try {
+    if (validationErrors.isEmpty()) {
+      const clients = await db.getActiveClients()
+      const message = req.body.message
+
+      for (const c of clients) {
+        // define phone_numbers as a Set to prevent duplicate numbers
+        const phone_numbers = new Set()
+
+        // add responder, fallback, and heartbeat numbers to phone_numbers
+        c.responderPhoneNumbers.forEach(pn => {
+          phone_numbers.add(pn)
+        })
+        c.fallbackPhoneNumbers.forEach(pn => {
+          phone_numbers.add(pn)
+        })
+        c.heartbeatPhoneNumbers.forEach(pn => {
+          phone_numbers.add(pn)
+        })
+
+        // for each phone number of this client
+        for (const pn of phone_numbers) {
+          // send SMS text message
+          twilioHelpers.sendTwilioMessage(pn, c.fromPhoneNumber, message)
```

Comment:
My logic here is that if the Twilio message fails, that's not an error on the part of this function, but rather the information stored in the database, or with the helper library / API keys stored in the environment. This is why I don't check for those occurrences in testing, or in the API call itself.

I also feel that in the case that a phone number is incorrect, it should be ignored and the function should proceed to the next client. In the case that ALL Twilio calls fail, then I feel that is a problem that is wider than this function, and I'd hope it would be apparent elsewhere...

@tmackinnon @ahmaddaadaa, because Theresa is no longer working full time, could you both give me your opinions on the above? Essentially the question here is: should there be checks in place to see if the `sendTwilioMessage` function fails, or simply call it with the client information available and let it resolve by itself (regardless of success or failure)?

Reply:
That said, I'll add `await` before the function call so that the server does not spam the Twilio API with message requests too fast.

Reply:
Hi Micah, I totally understand what you're saying about the scope of this function and initially agree with your point. But after some reflection, I have two further thoughts on this. 

1) I'm wondering if there's ever a reason that an outgoing twilio message wouldn't deliver even when the phone number in our database is correct and works (whether that's a twilio issues, a cell carrier issue, etc.)? If this is the case, then having accurate data in our db doesn't help us, and we should try and capture this somewhere so we know if someone did not receive our communication. 

2) Although it's not in the scope of this function, it may be valuable to know what phone numbers are not receiving messages from us. And would help us be proactive in ensuring the devices are working as they should (if we know a phone number isn't receiving messages then we can reach out to the client to fix this asap, as opposed to waiting for the client to eventually notice and potentially have missed a life-saving notification)

Let me know what your thoughts are - I'm happy to discuss further. :)

Reply:
After some investigation, I've found that in the event that the `sendTwilioMessage` function fails, it calls `helpers.logError(...)` from within the function; so this is already being kept track on the server!

See https://github.com/bravetechnologycoop/brave-alert-lib/blob/7773aca576752a29d3244c75b8764e8c1e51ef3e/lib/twilioHelpers.js#L32

Reply:
Ahh perfect thanks Micah! I think that should be fine then (although I'm still curious what Theresa's initial thinking was). @ahmaddaadaa I'm curious about your thoughts on this as well.

Reply:
Oh that's a great idea, I really like that!! Is our expectation that this will like be called from PA? That makes a lot of sense to me.

Reply:
Yes! Plan is that this will be implemented and called from PA. That's where the "Brave Key" originates from; `PA_API_KEY_PRIMARY` in the `.env` file.

Reply:
Awesome, I can't wait!

Reply:
This turned out great. Good job.